### PR TITLE
[Menubar] Improve keyboard controls

### PR DESCRIPTION
**Pull request by andy-hook**
- Add keyboard controls
- Add RTL support
- Tabbing is not possible yet per our other menu implementations

https://user-images.githubusercontent.com/11708259/208924409-4158de06-1974-4e5a-917b-18859ea6b583.mp4

### CODE REVIEW

**Comment by andy-hook** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1846** <a href="https://app.graphite.dev/github/pr/radix-ui/primitives/1846" target="_blank">Graphite</a> 
    * **PR #1847** <a href="https://app.graphite.dev/github/pr/radix-ui/primitives/1847" target="_blank">Graphite</a> 
      * **PR #1852** <a href="https://app.graphite.dev/github/pr/radix-ui/primitives/1852" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/radix-ui/primitives/1852?utm_source=stack-comment).

File: packages/react/roving-focus/src/RovingFocusGroup.tsx [hunk 1]
```diff
@@ -201,14 +201,22 @@ const ITEM_NAME = 'RovingFocusGroupItem';
...
-    const id = useId();
+    const {
+      __scopeRovingFocusGroup,
+      focusable = true,
+      active = false,
+      tabStopId,
+      ...itemProps
+    } = props;
+    const autoId = useId();
+    const id = tabStopId || autoId;
```
Comment (andy-hook) [inline packages/react/roving-focus/src/RovingFocusGroup.tsx:219]: I needed to control the tab stop as it's possible for the trigger to never receive focus in some cases.

I may have missed something but I couldn't find a way to provide my own ids, this made the existing `defaultCurrentTabStopId` prop confusing / useless?

File: packages/react/menubar/src/Menubar.tsx [hunk 2]
```diff
@@ -242,9 +325,50 @@ const MenubarContent = React.forwardRef<MenubarContentElement, MenubarContentPro
...
+          if (!menuContext.wasKeyboardTriggerOpenRef.current) event.preventDefault();
+        }}
+        onKeyDown={composeEventHandlers(
+          props.onKeyDown,
+          (event) => {
+            if (['ArrowRight', 'ArrowLeft'].includes(event.key)) {
+              const target = event.target as HTMLElement;
+              const targetIsSubTrigger = target.hasAttribute('aria-haspopup');
+              const isKeyDownInsideRoot =
+                target.closest('[data-radix-menu-content]') === event.currentTarget;
```
Comment (andy-hook) [inline packages/react/menubar/src/Menubar.tsx:346]: These two selections are controversial as we're relying on another components implementation details. I do think there is value here though as it allows us to colocate the entirely of this logic here, would providing attrs specific to this component alleviate some of the concerns?

Comment (benoitgrelard) [reply packages/react/menubar/src/Menubar.tsx:346]: > These two selections are controversial as we're relying on another components implementation details. I do think there is value here though as it allows us to colocate the entirely of this logic here,

I agree with the premise.

> would providing attrs specific to this component alleviate some of the concerns?

Given what we've recently gone through with [this](https://github.com/radix-ui/primitives/pull/1723) I'd say yes.

**Review by benoitgrelard — CHANGES_REQUESTED** [review]

File: packages/react/menubar/src/Menubar.tsx [hunk 3]
```diff
@@ -341,21 +350,22 @@ const MenubarContent = React.forwardRef<MenubarContentElement, MenubarContentPro
...
 
               const prevMenuKey = context.dir === 'rtl' ? 'ArrowRight' : 'ArrowLeft';
               const isPrevKey = prevMenuKey === event.key;
               const isNextKey = !isPrevKey;
 
+              // Prevent navigation when we're opening a submenu
+              if (isNextKey && targetIsSubTrigger) return;
+              // or we're inside a submenu and are moving backwards to close it
+              if (isKeyDownInsideSubMenu && isPrevKey) return;
+
```
Comment (benoitgrelard) [inline packages/react/menubar/src/Menubar.tsx:365]: Spot on now 👌 , love it!

**Review by benoitgrelard — APPROVED** [review]

**Pull request merged**